lightningd/log.c

```diff
@@ -55,12 +55,34 @@ struct log_book {
 struct log {
 	struct log_book *lr;
 	const struct node_id *default_node_id;
-	const char *prefix;
+	struct log_prefix *prefix;
 
 	/* Non-NULL once it's been initialized */
 	enum log_level *print_level;
 };
 
+static struct log_prefix *log_prefix_new(const tal_t *ctx,
+					 const char *prefix TAKES)
+{
+	struct log_prefix *lp = tal(ctx, struct log_prefix);
+	lp->refcnt = 1;
+	lp->prefix = tal_strdup(lp, prefix);
+	return lp;
+}
+
+static void log_prefix_drop(struct log_prefix *lp)
+{
+	if (--lp->refcnt == 0)
+		tal_free(lp);
+}
```

Comment:
Looks like Valgrind considered this as invalid behavior?

Reply:
Yes, and it's right: access after free. We need to do this a bit differently, will address tomorrow!

Reply:
For sure this is a very stupid question, but I'm missing the point of the code that makes valid happy again.

I know, it is stupid (maybe)